### ISSUE TITLE
Links to Languages/JS videos broken

### ISSUE THREAD

**Issue by anthonyisensee** [microsoft/vscode-docs#5717]
Videos at https://code.visualstudio.com/docs/languages/javascript lead to broken links. Have tested on multiple browsers and networks to ensure it's not a me problem.

For example, both of the broken links are rendered to the page:

- /assets/docs/languages/docs/languages/javascript/intellisense.mp4
- /assets/docs/languages/docs/languages/javascript/snippets.mp4

Based on examining `.gif` files on the R page it appears they should lead to these links (boom, there the files are!):

- /assets/docs/languages/javascript/intellisense.mp4
- /assets/docs/languages/javascript/snippets.mp4

In other words, the extra "docs/languages/" must be removed from the links.

**Comment by anthonyisensee** [comment]
https://raw.githubusercontent.com/microsoft/vscode-docs/main/docs/languages/javascript.md currently contains the following:

- /docs/languages/javascript/intellisense.mp4
- /docs/languages/javascript/snippets.mp4

**Comment by anthonyisensee** [comment]
Based on the assumption that something is adding `/assets/docs/languages` when it only sees `/docs/languages/javascript/file.mp4` I've created [a pull request](https://github.com/microsoft/vscode-docs/pull/5718) to use absolute links (as explained [here](https://github.com/microsoft/vscode-docs/blob/c41175e8775cde38ae641153848a6ce26a0dae9c/CONTRIBUTING.md#links) in contribution guidelines). I hope I am not misunderstanding how website build works, but I may be.

**Comment by gregvanl** [comment]
Hi @anthonyisensee Thanks for pointing this out. This seems odd and I had a look at previous versions of the website via the WayBack Machine website and those paths had been working back in 2020. Not sure yet what changed.
Also odd is that some topics have no problem rendering similar video paths. For example, https://code.visualstudio.com/docs/languages/java uses:

<video autoplay loop muted playsinline controls>
  <source src="/docs/languages/java/package-viewer.mp4" type="video/mp4">
</video>

**Comment by anthonyisensee** [comment]
Indeed, that is odd. 

Both

- https://code.visualstudio.com/assets/docs/languages/java/package-viewer.mp4
and
- https://code.visualstudio.com/docs/languages/java/package-viewer.mp4

lead to the same file despite a completely different path. The same behavior also exists with the javascript files.

Now that I'm examining the HTML a bit more closely, there may be a major HTML difference here.

javascript page:

```html
<video src="/docs/languages/javascript/snippets.mp4" placeholder="images/javascript/snippets-placeholder.png" autoplay loop controls muted>
    Sorry, your browser doesn't support HTML 5 video.
</video>
```

java page: 

```html
<video autoplay loop muted playsinline controls>
  <source src="/docs/languages/java/package-viewer.mp4" type="video/mp4">
</video>
```

One `<video>` element contains a `<source>` element while the other instead uses a `src` attribute on the video tag. I believe these are functionally equivalent in HTML (see [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#src)) but does whatever is doing your web build know how to (or not to) handle re-formatting of routes on a `<video>` elements `src` attribute?

Chasing straws. Especially curious since you said the same links were working on WayBack.

**Comment by gregvanl** [comment]
Hi @, I think I've got it sorted and I also think I've got a preferred solution. 
Maybe you already noticed that the placeholder image paths are preprocessed and end up correct on the website. For example, the relative path in the example above 

```html
placeholder="image/javascript/snippet-placeholder.png"
```
becomes:

![image](https://user-images.githubusercontent.com/12818376/196592564-7e5c8499-bcf1-4cfc-ac88-642567586edd.png)

This is because the website build fixes up the path for `image/` paths.

I think the best fix would be to update the src path to be the relative vscode-docs path also "images/javascript/snippets.mp4"

I don't like the `<source src=` solution or hardcoding `/assets` since these require knowledge about the final website asset locations.

Would you like to update your PR to do that change? Can I also ask you to fix two TypeScript language files as well as they have the same issue:

/docs/typescript/typescript-editing.md
/docs/typescript/typescript-refactoring.md

In summary, the fixes would look like:

![image](https://user-images.githubusercontent.com/12818376/196593287-d25aa063-a4ab-4a8b-830a-f96fbc3a5587.png)

**Comment by anthonyisensee** [comment]
@gregvanl sure thing. I'm working off of only what I can guess of your web build and how it handles preparing links and file routing, however, so do the typescript files links go from being:

`/docs/typescript/editing/jsdoc-autofill.mp4`
to
`images/typescript/editing/jsdoc-autofill.mp4`
OR
`images/typescript/jsdoc-autofill.mp4` ?

Let me know and I'll make the changes.

**Comment by gregvanl** [comment]
It is a little tricky. Remove `/docs/typescript` and replace with `images`.
The path should match the relative path in the images folder.

![image](https://user-images.githubusercontent.com/12818376/196758068-5bc95043-fe45-47ca-9eef-0f01a5b1aa6a.png)

So the final path should be `images/editing/jsdoc-autofile.mp4`.

Really appreciate you investigating and pursuing this!

**Comment by anthonyisensee** [comment]
Certainly. Anything for (the docs of) my weapon of choice.

See [pull request](https://github.com/microsoft/vscode-docs/pull/5718 with updates) updates.

**Comment by gregvanl** [comment]
Published shortly and looks good on our staging server.
Again thanks for working on this!

**Comment by anthonyisensee** [comment]
Huzzah! It works then. Fantastic.

Certainly, my pleasure.
